### PR TITLE
Add operator MergeFragmentedActivityOperator

### PR DESCRIPTION
**Pull request by abalhomaid**
Related to Interns notebook

### CODE REVIEW

**Review by uabbas — DISMISSED** [review]

**Comment by abalhomaid** [comment]
I ran through the interns code, and tested some cases. I understood how the merging algorithm works in details, but in some cases, I'm not sure whether the sleep should be merged or not.
I am discussing this with Joao very soon

**Comment by abalhomaid** [comment]
Ready to be reviewd

File: tasrif/test_scripts/MergeFragmentedActivityOperator.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# ---
```
Comment (mak2508) [inline tasrif/test_scripts/MergeFragmentedActivityOperator.py:1]: Consider renaming this file to test_pipeline_MergeFragmentedActivityOperator.py
To be more consistent with other tests.

File: tasrif/processing_pipeline/custom/merge_fragmented_activity_operator.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+class MergeFragmentedActivityOperator(ProcessingOperator):
+
+    """
+    Operator to merge gaps between activity. A gap is a time period of activity less than a given threshold.
+    A period of activity that is larger than threshold is considered the end of an activity.
+    Gaps found before the end of the activity will be merged with the activity.
+    The end date of the activity will be set to the merged `end_date_feature_name`
+    The input of this operator needs to have two datetime columns that represent the starting time of the
+    activity and end time of the activity, respectively.
```
Comment (mak2508) [inline tasrif/processing_pipeline/custom/merge_fragmented_activity_operator.py:18]: I think this explanation of the Operator is not very clear. It took me quite some time to understand what the operator is doing.

I think just specifying that the gaps are identified by comparing the end time of an activity with the following activity's start time would make this much more clear.
Maybe something like:

`Operator to merge gaps between activity. A gap is a time period of activity less than a given threshold.
    A period of activity that is larger than threshold is considered the end of an activity.
    Gaps found before the end of the activity will be merged with the activity.
    A gap is identified by comparing the end time of an activity and comparing it with the start time of the following activity.
    The end date of the activity will be set to the merged `end_date_feature_name`
    The input of this operator needs to have two datetime columns that represent the starting time of the
    activity and end time of the activity, respectively.`

Comment (abalhomaid) [reply tasrif/processing_pipeline/custom/merge_fragmented_activity_operator.py:18]: Thank you Muhammad for the detailed explination

Comment (mak2508) [reply tasrif/processing_pipeline/custom/merge_fragmented_activity_operator.py:18]: You're welcome!

**Review by uabbas — APPROVED** [review]

**Review by mak2508 — APPROVED** [review]

**Pull request merged**